### ISSUE TITLE
Fix behavior when journal not publicly enabled

### ISSUE THREAD

**Issue by NateWr** [pkp/pkp-lib#592]
There are a few issues with the publicly enabled journal setting.
- [x] The setting appears under **Settings > Journal > Masthead** as well as under **Admin > Hosted Presses**. Seems like it should only appear under Admin.
- [x] To ensure Google (and other search engines) do not inadvertently index a site that is still being developed, we should add `noindex` and `nofollow` meta tags to the HTML `head` of a site if it has not been enabled in Journal Manager settings: `<meta name="robots" content="noindex,nofollow">`. See [this thread](http://forum.pkp.sfu.ca/t/google-indexing-non-public-journals/1914) in the forums.
- [x] Test whether or not this actually removes it from the site-wide journal index when multiple journals exist.
- [ ] When only one journal exists, and it's not set to be publicly enabled, the site index URL will still automatically redirect to the journal. Perhaps it shouldn't.

**Comment by NateWr** [comment]
The following PR adds a `noindex,nofollow` meta tag when journals are not enabled:
https://github.com/pkp/pkp-lib/pull/1646

The last checkbox is still open. I went to turn that off, so it doesn't redirect to a journal that's not publicly enabled. That code is here:

https://github.com/pkp/ojs/blob/master/pages/index/IndexHandler.inc.php#L36-L46

I realized, though, that disabling this might make it impossible to reach a login page. Do we have any isolated login area that could be reached if all journals are not publicly enabled? If not, we'll have to leave things as they are for now and defer the issue until someone complains.

(Also, in the code linked above, there's a `$journalsCount` variable that doesn't seem to be defined anywhere. Is this code working?)

**Comment by asmecher** [comment]
`$journalsCount` comes back from `getTargetContext` by reference. It's a weird optimization.

**Comment by NateWr** [comment]
Assigning to 3.0.1 milestone as the main issues are resolved.

**Comment by NateWr** [comment]
PRs:
https://github.com/pkp/pkp-lib/pull/1916
https://github.com/pkp/ojs/pull/1075

Tests running.

This takes a pretty low-level approach to completely removing the visibility of disabled journals. Requests to `/index.php/single` will result in a 404 response to any logged-out users. The site index is still available for logging in.

**Comment by asmecher** [comment]
@bozana, would you mind taking a look at this?

**Comment by bozana** [comment]
@NateWr, maybe one more point/checkbox could be added to this issue: what about the option "Which journals on this site would you like to register with?" on the registration page?

**Comment by NateWr** [comment]
Ok, I've rebased the PRs and added one for OMP. Tests are re-running.

I checked the journalists listed for registration on the site index registration page and disabled journals are already excluded. Good thinking! :+1:

**Comment by bozana** [comment]
Great @NateWr! Everything merged. Can this issue now be closed? THANKS!

**Comment by bozana** [comment]
Or should it be cherry-picked to ojs-stable-3_0_0, omp_dev_1_2 and omp-stable_1_2_0?

**Comment by NateWr** [comment]
To be honest, I'm not really sure what should and shouldn't be merged to those branches. Are they getting all new features for the time being?

**Comment by NateWr** [comment]
@asmecher should this be cherry-picked to ojs-stable-3_0_0, omp_dev_1_2 and omp-stable_1_2_0?

**Comment by bozana** [comment]
I always think like this: if it is a bug fix that the current users of ojs-stable-3_0_0 or omp-stable-1-2_0 could use, then cherry-pick it to that brunches. If it is a 'smaller' and well tested change or feature, that would be good to go into the next OMP release, cherry-pick it to omp-dev-1_2. But...

**Comment by bozana** [comment]
Hmmm.. also, the next small releases could be made using the stable branches, so maybe, as Alec said here https://github.com/pkp/pkp-lib/issues/1709#issuecomment-257439133: 

> does this have compatibility-breaking potential? If so, don't back-port to omp-stable-1_2_0, just to omp-dev-1_2. If it's just bugfixes and new capabilities, then it's good to back-port to omp-stable-1_2_0 too.

**Comment by asmecher** [comment]
Yes, I think these are low-risk enough to justify back-ports.

**Comment by NateWr** [comment]
Cherry-picked. Closing.

**Comment by isgrim** [comment]
Hi everyone,

I found some weird behaviour that came up, while debugging a 404 Error with our OMP instance (using branch omp-stable-1_2_0). A press was accidentally disabled and that lead to 404 errors for requests to the login page of this press as a logged out user.
I traced the error to changes made for this issue, in the PKPPageRouter class:
https://github.com/pkp/pkp-lib/blob/70849ee5dd1e21c5baa3cb69a7beea70d95ebbd1/classes/core/PKPPageRouter.inc.php#L179-L205
With the changes the router should route to the default index, but it returns with a 404 error, because the path to sourceFile for the index page is not correctly set, during the debug session it was set to "pages/./pages/index/index.php/index.php". The file does not exist an so the `$dispatcher->handle404();` is called.

Is this is the intended behaviour or a bug?

Thanks for your time and a happy new year,

Nils
